samples/javascript_nodejs/21.corebot-app-insights/index.js

```diff
@@ -0,0 +1,96 @@
+const { BotFrameworkAdapter, ConversationState, InputHints, MemoryStorage, NullTelemetryClient, UserState } = require('botbuilder');
+const { FlightBookingRecognizer } = require('./dialogs/flightBookingRecognizer');
+
+// This bot's main dialog.
+const { DialogAndWelcomeBot } = require('./bots/dialogAndWelcomeBot');
+const { MainDialog } = require('./dialogs/mainDialog');
+const { BookingDialog } = require('./dialogs/bookingDialog');
+
+// Note: Ensure you have a .env file and include LuisAppId, LuisAPIKey and LuisAPIHostName.
+const ENV_FILE = path.join(__dirname, '.env');
+require('dotenv').config({ path: ENV_FILE });
+
+// Create adapter.
+// See https://aka.ms/about-bot-adapter to learn more about adapters.
+const adapter = new BotFrameworkAdapter({
+    appId: process.env.MicrosoftAppId,
+    appPassword: process.env.MicrosoftAppPassword
+});
+
+// Catch-all for errors.
+adapter.onTurnError = async (context, error) => {
+    // This check writes out errors to console log
+    // NOTE: In production environment, you should consider logging this to Azure
+    //       application insights.
+    console.error(`\n [onTurnError]: ${ error }`);
+    // Send a message to the user
+    const onTurnErrorMessage = 'Sorry, it looks like something went wrong!';
+    await context.sendActivity(onTurnErrorMessage, onTurnErrorMessage, InputHints.ExpectingInput);
+    // Clear out state
+    await conversationState.delete(context);
+};
+
+// Define a state store for your bot. See https://aka.ms/about-bot-state to learn more about using MemoryStorage.
+// A bot requires a state store to persist the dialog and user state between messages.
+
+// For local development, in-memory storage is used.
+// CAUTION: The Memory Storage used here is for local bot debugging only. When the bot
+// is restarted, anything stored in memory will be gone.
+const memoryStorage = new MemoryStorage();
+const conversationState = new ConversationState(memoryStorage);
+const userState = new UserState(memoryStorage);
+
+// If configured, pass in the FlightBookingRecognizer.  (Defining it externally allows it to be mocked for tests)
+const { LuisAppId, LuisAPIKey, LuisAPIHostName } = process.env;
+const luisConfig = { applicationId: LuisAppId, endpointKey: LuisAPIKey, endpoint: `https://${ LuisAPIHostName }` };
+
+const luisRecognizer = new FlightBookingRecognizer(luisConfig);
+
+const telemetryClient = getTelemetryClient(process.env.InstrumentationKey);
+
+// Create the main dialog.
+const bookingDialog = new BookingDialog(telemetryClient);
+const dialog = new MainDialog(luisRecognizer, bookingDialog, telemetryClient);
+const bot = new DialogAndWelcomeBot(conversationState, userState, dialog);
+
+// Create HTTP server
+const server = restify.createServer();
+
+// Enable the Application Insights middleware, which helps correlate all activity
+// based on the incoming request.
+// server.use(restify.plugins.bodyParser());
```

Comment:
Hmm - why is this commented out?  I believe this is required for user_id and session_id to be populated properly..
server.use(restify.plugins.bodyParser());

Reply:
Hi @daveta, we just uncommented this line.